api/src/license_magic.rs

```diff
@@ -0,0 +1,34 @@
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub struct LicenseMagic {
+    magic: Vec<Vec<u8>>,
+}
+
+impl LicenseMagic {
+    pub fn new(magic: Vec<Vec<u8>>) -> LicenseMagic {
+        LicenseMagic { magic }
+    }
+
+    pub fn default() -> LicenseMagic {
+        LicenseMagic { magic: Vec::new() }
+    }
+
+    pub fn push(&mut self, magic: Vec<u8>) -> &Self {
```

Comment:
Well, I planned to use push just as chained function, 

for example:

```rust
LicenseOperator::default(...).push(...)...
```

but I don't actually need that, so I removed it and return nothing. And anyway from my current state, user will not have access to `magic` because it is private field in LicenseOperator and I don't have any getters for it, so it was useless...

And yes without `mut` it is bad design...